pkg/proc/disasm_amd64_arch.go

```diff
@@ -195,3 +197,47 @@ func checkPrologue(s []AsmInstruction, prologuePattern instrseq) bool {
 	}
 	return true
 }
+
+func disassemble(memrw MemoryReadWriter, regs Registers, breakpoints *BreakpointMap, bi *BinaryInfo, startAddr, endAddr uint64, singleInstr bool) ([]AsmInstruction, error) {
```

Comment:
As far as I can tell the only difference between this and the disassemble in disasm_amd64 is the number of bytes advanced for an unknown instruction (1 here, 4 for arm64). I think this function should be left inside disasm.go and take an `arch Arch` argument. Then:

1. Add a `MinInstructionLength` member to `type Arch interface`
2. Define `func (a *AMD64) MinInstructionLength() int { return 1 }` and `func (a *ARM64) MinInstructionLength() int { return 4 }`
3. Call MinInstructionLength in this function.